scripts/setup_conan.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# Copyright (C) 2025 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+#
+
+conan_option=${CONAN_MAJOR_VERSION:-2}
+
+# Since this script is installed with the name "conan", there is a risk of infinite recursion if pyenv is not available
+# on the PATH, which can occur when setting up a development environment. To prevent this, check for recursive calls.
+if [[ "$CONAN_RECURSIVE_CALL" -eq 1 ]]; then
```

Comment:
A simpler check could be to compare `${BASH_SOURCE[1]}` with this script's path, i.e. `${BASH_SOURCE[0]}`.